lib/orocos/process.rb

```diff
@@ -1171,14 +1183,32 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
                     block.call(process)
                 else
                     # Get any task name from that specific deployment, and check we
-                    # can access it. If there is none
+                    # can access it either through the ior pipe or through the CORBA name
+                    # service.
+                    if process.alive?
+                        begin
+                            ior_message += ior_read.read_nonblock 4096
+                        rescue IO::WaitReadable
+                        end
+
+                        if ior_read.eof?
+                            if ior_message_valid? process, ior_message
+                                process.task_names.each do |name|
+                                    process.register_ior name, message[name]
+                                end
+                            else
+                                raise Orocos::InvalidIORMessage,
+                                      "the received ior message is invalid. Received: #{ior_message}"
+                            end
+                        end
+                    end
+
                     all_reachable = process.task_names.all? do |task_name|
-                        if name_service.task_reachable?(task_name)
-                            Orocos.debug "#{task_name} is reachable"
+                        if process.has_ior_for? task_name
+                            Orocos.debug "#{task_name}'s IOR is registered"
                             true
                         else
-                            Orocos.debug "could not access #{task_name}, #{name} is not running yet ..."
-                            false
+                            Orocos.debug "No IOR is registered for #{task_name}"
```

Comment:
This was necessary because we have no explicit communication with the deployment. Now that it is the deployment sending us the data, not having the IOR of a task is a discrepancy between our side (that expects a list of tasks) and the actual process, and that should definitely be handled as an error

In other words, calculate all_reachable only if we have read the complete message (i.e. eof? and stuff), and raise if a task is missing. That is, remove all all_reachable logic from this method, rename `ior_message_valid?` into `load_and_validate_ior_message` and raise from there if a task is missing.